.github/workflows/smoketest.yml

```diff
@@ -0,0 +1,95 @@
+on:
+  pull_request:
+  merge_group:
+name: SmokeTest
+concurrency:
+  group: ${{ github.ref }}-smoketest
+  cancel-in-progress: true
+jobs:
+  smoke-test-upgrade-path:
+    name: Smoke Test Upgrade Path
+    # if: github.event_name != 'pull_request' || github.event.action == 'enqueued' || contains( github.event.pull_request.labels.*.name, 'run-all')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository with full history
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+          fetch-tags: true
+      - name: Get last tagged release
+        run: |
+          latest_release=$(git tag --sort=-v:refname | grep -v v1 | head -n 1)
+          echo "Latest release: $latest_release"
+          echo "LATEST_VERSION=${latest_release#v}" >> "$GITHUB_ENV"
+      - name: Check out the last tagged release
+        uses: actions/checkout@v4
+        with:
+          ref: v${{ env.LATEST_VERSION }}
+          fetch-depth: 1
+          fetch-tags: true
+      # The following grabs the current smoke test and deployment Justfile from the HEAD. This
+      # is to solve the chicken-and-egg. Once it's in the tagged release, skip this.
+      - name: Check out the current smoke test
+        uses: actions/checkout@v4
+        with:
+          clean: false
+          repository: TBD54566975/ftl.git
+          path: tmpsmoketest
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+          sparse-checkout: |
+            smoketest
+            deployment
+          sparse-checkout-cone-mode: false
+      # When we remove the above step to fetch the current smoke tests and deployment Justfile
+      # into tmpsmoketest, remove the overwrite step as well.
+      - name: Replace the tagged release smoketest with the current smoketest
+        run: |
+          set -euo pipefail
+          echo "Replacing tagged release smoketest with current smoketest"
```

Comment:
This approach will break if the smoke test adds new features that are in the new version but not the old version, as you are testing the new code again old FTL.

This is fine for now but I think but will need some thought going forward.